clcache.py

```diff
@@ -232,6 +232,9 @@ def getManifestHash(compilerBinary, commandLine, sourceFile):
         # One of the few exceptions to this rule is the /MP switch, which only
         # defines how many compiler processes are running simultaneusly.
         commandLine = [arg for arg in commandLine if not arg.startswith("/MP")]
+        baseDir = os.environ.get('CLCACHE_BASEDIR')
+        if baseDir:
+            commandLine = [arg.replace(baseDir, BASEDIR_REPLACEMENT) for arg in commandLine]
```

Comment:
Ah, true - as it is, `collapseBasedirToPlaceholder` is not applicable. While thinking about whether it makes sense to adjust it as you suggested, it occurred to me that maybe clcache shouldn't always blindly replace the basedir value.

For instance, should it really replace the basedir in preprocessor definitions passed via `-D` on the command line?